src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -759,7 +757,10 @@ ClientResponse<byte[], byte[]> run() {
         .setHeader("Content-Type", contentType != null ? (contentType + (characterEncoding != null ? "; charset=" + characterEncoding : "")) : null)
         .setHeaders(request.getHeaders())
         .method(request.getMethod().name())
-        .readTimeout(0)
+        // Zero should be ok, but under load I believe we are getting a timeout that can lead to a truncated body.
+        // And since we are streaming the response, the preamble will already have been written which means the status code
+        // will indicate success. So I think a 2ms timeout is reasonable to avoid this type of situation.
+        .readTimeout(2)
```

Comment:
0 is interpreted as infinite timeout.  This could actually timeout a lot of reads.
https://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#setReadTimeout-int-

Reply:
I'm an idiot. Thanks.